src/WebJobs.Script/Extensions/IDictionaryExtensions.cs

```diff
@@ -19,5 +20,27 @@ public static class IDictionaryExtensions
                 dictionary[pair.Key] = pair.Value;
             }
         }
+
+        public static IDictionary<string, TValue> AsCaseInsensitive<TValue>(this IDictionary<string, TValue> dictionary)
+        {
+            return new Dictionary<string, TValue>(dictionary, StringComparer.OrdinalIgnoreCase);
```

Comment:
Removed the whole case insensitive thing. Modified TryGetValue helper to support case insensitive queries, which matches the logic we had before with JObject